### ISSUE TITLE
rsync child process being killed immediately 

### ISSUE THREAD

**Issue by hurngchunlee** [Donders-Institute/data-streamer#7]
In modalityMEG.js, the 'rsync' process of the 'meg_copy.sh' called via child_process.spawn can eventually being killed by the system (kernel?) after few days of operation.

Restarting the streamer resolved the problem.  The killing seems to be due to "running out of resource (memory in this case?)".

**Comment by hurngchunlee** [comment]
Along the line of debugging this issue, there seems to be memory leak in the 'modalityMRI.js'.  The cause is unclear, but could be due to the "Promise" object returned from the orthanc-client library. 

For the time being, the streamer is instructed with 'heap dump' and an regular `gc` call in the worker process.  

Attached are heap dump from before and after the MRI job is executed:
- [heapdump-beforeMriJob.heapsnapshot.gz](https://github.com/Donders-Institute/streamer/files/636095/heapdump-beforeMriJob.heapsnapshot.gz)
- [heapdump-afterMriJob.heapsnapshot.gz](https://github.com/Donders-Institute/streamer/files/636096/heapdump-afterMriJob.heapsnapshot.gz)

**Comment by hurngchunlee** [comment]
Another observation: looks like the issue happens in the callback of the stdout/stderr on the 'data' event.  Putting the following code in meg_copy.sh (before rsync is called) can also reproduce the issue:

```bash
for i in {1..100}; do
    echo $i
    echo $i 1>&2
    sleep 1
done
```

Later on, I also noticed that it happens to one worker which has relatively higher 'rss'.

**Comment by hurngchunlee** [comment]
Something I just learned about the way Node.js frees up the memory allocated for Buffer objects: it's not entirely under the control of the V8 engine.

The effect of it is that in the `process.memoryUsage()`, the rss (resident set size) is growing over the time.  The impact is more significant when downloading many DICOM images files in parallel from the Orthanc server using the [orthanc-client](https://www.npmjs.com/package/orthanc-client).  This is because the `instances.getFile` function uses the `Buffer` object to store the image data.  

This page described the similar issue, and their solution with IO piping: https://sankhs.com/2016/03/31/malloc-and-rss-memory/

I created an [example script](https://github.com/Donders-Institute/streamer/blob/master/docker/streamer/scripts/oc_getfile_nobuffer.js) using the simple http client to download the data file, and `pipe` the server response (the image data) to a local file.
